sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -256,6 +256,7 @@ public abstract partial class Response<T> : Azure.NullableResponse<T>
         protected Response() { }
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override bool HasValue { get { throw null; } }
+        public override T Value { get { throw null; } }
```

Comment:
Curious: what is the benefit of overriding this as non-nullable if it can still have a null value?

Reply:
The type is intended to communicate that this should never be null. Although the compiler won't disallow null values, making it nullable would be counter to that intent. Essentially, the type is the same as it was before this change, but since the base type property changed to nullable, this override is necessary to keep it as it was before.

Reply:
> The type is intended to communicate that this should never be null.

But HasValue is being updated to allow for it being null.

Reply:
This is only for back compat since we have already shipped libraries that set this value to null, with the intent of it meaning there is no value.

Reply:
Sure, but there is nothing in the property that says this is only for back compat. This is the new behavior going forward so `Response<T>` is totally functionally, AFAICT, for use with null response values.

Reply:
As mentioned [here](https://github.com/Azure/azure-sdk-for-net/pull/35560#discussion_r1242700869) this is the current and intended nullability of the type. To change it to nullable would be worse, IMO.